### ISSUE TITLE
Open SSL Issue with Python 3.11 docker container

### ISSUE THREAD

**Issue by g-tyagi** [Azure-Samples/cognitive-services-speech-sdk#2398]
- Speech SDK log taken from a run that exhibits the reported issue. 
`{
    "_RecognitionResult__handle": {
      "_Handle__handle": {},
      "_Handle__test_fn": {
        "__name__": "recognizer_result_handle_is_valid"
      },
      "_Handle__release_fn": {
        "__name__": "recognizer_result_handle_release"
      }
    },
    "_offset": 0,
    "_duration": 0,
    "_result_id": "db27961b77a842979b47a21707fc2354",
    "_reason": 1,
    "_text": "",
    "_propbag": {
      "_PropertyCollection__handle": {
        "_Handle__handle": {},
        "_Handle__test_fn": {
          "__name__": "property_bag_is_valid"
        },
        "_Handle__release_fn": {
          "__name__": "property_bag_release"
        }
      }
    },
    "_json": "",
    "_error_json": "Connection failed (no connection to the remote host). Internal error: 1. Error details: Failed with error: WS_OPEN_ERROR_UNDERLYING_IO_OPEN_FAILED\nwss://eastus2.stt.speech.microsoft.com/speech/universal/v2\nX-ConnectionId: 9879770ce56347b6bde0278535b7b1c1 SessionId: 9879770ce56347b6bde0278535b7b1c1",
    "_properties": {
      "5001": "Connection failed (no connection to the remote host). Internal error: 1. Error details: Failed with error: WS_OPEN_ERROR_UNDERLYING_IO_OPEN_FAILED\nwss://eastus2.stt.speech.microsoft.com/speech/universal/v2\nX-ConnectionId: 9879770ce56347b6bde0278535b7b1c1 SessionId: 9879770ce56347b6bde0278535b7b1c1"
    },
    "_cancellation_details": {
      "_CancellationDetails__reason": 1,
      "_CancellationDetails__code": 5,
      "_CancellationDetails__error_details": "Connection failed (no connection to the remote host). Internal error: 1. Error details: Failed with error: WS_OPEN_ERROR_UNDERLYING_IO_OPEN_FAILED\nwss://eastus2.stt.speech.microsoft.com/speech/universal/v2\nX-ConnectionId: 9879770ce56347b6bde0278535b7b1c1 SessionId: 9879770ce56347b6bde0278535b7b1c1"
    },
    "_no_match_details": null
  },`

**Setup to reproduce bug**

Building a docker container of a FastAPI app. App works fine itself but receive `WS_OPEN_ERROR_UNDERLYING_IO_OPEN_FAILED` when using dockerised image.
Added `ca-certificates` based on https://github.com/MicrosoftDocs/azure-docs/blob/main/articles/ai-services/speech-service/includes/quickstarts/platform/linux-distributions.md.
Confirmed openssl version on built image - 1.1.1f. Ref: https://github.com/Azure-Samples/cognitive-services-speech-sdk/issues/2048

**Dockerfile**
`FROM ubuntu:20.04
RUN apt-get --allow-releaseinfo-change update && apt-get install -y software-properties-common
RUN  add-apt-repository ppa:deadsnakes/ppa
RUN  apt-get install -y --no-install-recommends \
    python${PYTHON_VERSION} \
    python${PYTHON_VERSION}-dev \
    python3-venv \
    python3-pip \
    python${PYTHON_VERSION}-distutils -y

RUN apt-get install -y build-essential libssl-dev ca-certificates wget

RUN apt-get -y install curl
RUN curl https://bootstrap.pypa.io/pip/get-pip.py --output get-pip.py && \
    python${PYTHON_VERSION} get-pip.py

COPY requirements.txt tmp/
RUN python3.11 -m pip install -i hrequirements.txt

EXPOSE 8000
EXPOSE 80
EXPOSE 443`

**Platform, Operating System, and Programming Language**

 - OS: MacOS 14.4
 - Hardware - ARM
 - Programming language: Python

**Comment by ilirosmanaj** [comment]
Not sure if it helps, but here is the readme from one my repos (I think I faced the same issue a while ago). Hope it helps:

---

The above will install the python requirements from the app. Anyway, to run the Azure speech service, you will need 
to follow the instructions from the following link: [here](https://learn.microsoft.com/en-us/azure/ai-services/speech-service/quickstarts/setup-platform?pivots=programming-language-csharp&tabs=linux%2Cubuntu%2Cdotnetcli%2Cdotnet%2Cjre%2Cmaven%2Cnodejs%2Cmac%2Cpypi).

The short version is that:

* If you are running on Ubuntu, you need the latest LTS
* Speech SDK only supports OpenSSL v1
* Make sure to have the following environment variables exported
  * `SSL_CERT_DIR`
  * `LD_LIBRARY_PATH`
    * Make sure to export these variables in a permanent environment file, like `~/.bashrc` or `/etc/environment`

Just in case these values change, here they are:

```bash
export SSL_CERT_DIR=/etc/ssl/certs
export LD_LIBRARY_PATH=/usr/local/lib:$LD_LIBRARY_PATH
```

**Comment by g-tyagi** [comment]
VPN was appending a new root CA at end of cert chain. Works off VPN.
